### PR TITLE
Unit test for jazz_is-slack-channel-available

### PR DESCRIPTION
**Pull request by supritat**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### CODE REVIEW

File: core/jazz_is-slack-channel-available/index.js [hunk 1]
```diff
@@ -149,3 +148,12 @@ function getResponse(channel_url, channel_name) {
     });
   });
 }
+
+const exportable = {
```
Comment (svidhani) [inline core/jazz_is-slack-channel-available/index.js]: Thank you

File: core/jazz_is-slack-channel-available/test/test.js [hunk 2]
```diff
@@ -1,11 +1,282 @@
...
+    var channel_url ,channel_name;
+
+    beforeEach(() => {
+      channel_url = config.public_channel_endpoint;
+      channel_name = event.query.slack_channel;
+    });
+
+    it("should successfully search private slack channel in user's slack list", () => {
+      var resObj = {
+        body: "{\"ok\":true,\"channels\":[{\"id\":\"A1B2C3D4E\",\"name\":\"general\",\"is_channel\":true},{\"id\":\"A1B2C3D4F\",\"name\":\"jazz\",\"is_channel\":true},{\"id\":\"A1B2C3D4G\",\"name\":\"test-channel\",\"is_channel\":true},{\"id\":\"A1B2C3D4H\",\"name\":\"random\",\"is_channel\":true},{\"id\":\"A1B2C3D4I\",\"name\":\"test\",\"is_channel\":true}]}"
```
Comment (suryajak) [inline core/jazz_is-slack-channel-available/test/test.js]: Convert this to a readable JSON and then stringify to match the underlying components interface.

File: core/jazz_is-slack-channel-available/test/test.js [hunk 3]
```diff
@@ -1,11 +1,282 @@
...
+        .then(res => {
+          expect(res).to.eq('true');
+          sinon.assert.calledOnce(reqStub);
+          reqStub.restore();
+        });
+    });
+
+    it("should successfully search group slack channel in user's slack list", () => {
+      var resObj = {
+        body: "{\"ok\":true,\"groups\":[{\"id\":\"A1B2C3D4E\",\"name\":\"general\",\"is_channel\":true},{\"id\":\"A1B2C3D4F\",\"name\":\"jazz\",\"is_channel\":true},{\"id\":\"A1B2C3D4G\",\"name\":\"test-channel\",\"is_channel\":true},{\"id\":\"A1B2C3D4H\",\"name\":\"random\",\"is_channel\":true},{\"id\":\"A1B2C3D4I\",\"name\":\"test\",\"is_channel\":true}]}"
```
Comment (suryajak) [inline core/jazz_is-slack-channel-available/test/test.js]: Convert this to a readable JSON and then stringify to match the underlying components interface.

File: core/jazz_is-slack-channel-available/test/test.js [hunk 4]
```diff
@@ -1,11 +1,282 @@
...
+        .then(res => {
+          expect(res).to.eq('true');
+          sinon.assert.calledOnce(reqStub);
+          reqStub.restore();
+        });
+    });
+
+    it("should indicate false if slack channel is not available in user's slack list", () => {
+      var resObj = {
+        body: "{\"ok\":true,\"channels\":[{\"id\":\"A1B2C3D4E\",\"name\":\"general\",\"is_channel\":true},{\"id\":\"A1B2C3D4F\",\"name\":\"jazz\",\"is_channel\":true},{\"id\":\"A1B2C3D4H\",\"name\":\"random\",\"is_channel\":true},{\"id\":\"A1B2C3D4I\",\"name\":\"test\",\"is_channel\":true}]}"
```
Comment (suryajak) [inline core/jazz_is-slack-channel-available/test/test.js]: Convert this to a readable JSON and then stringify to match the underlying components interface.

File: core/jazz_is-slack-channel-available/test/test.js [hunk 5]
```diff
@@ -1,11 +1,282 @@
...
+        .then(res => {
+          expect(res).to.eq('false');
+          sinon.assert.calledOnce(reqStub);
+          reqStub.restore();
+        });
+    });
+
+    it("should indicate false if user's slack list is empty", () => {
+      var resObj = {
+        body: "{\"ok\":true,\"channels\":[]}"
```
Comment (suryajak) [inline core/jazz_is-slack-channel-available/test/test.js]: This is ok, but to keep it consistent, convert this to a readable JSON and then stringify to match the underlying components interface.

Comment (suryajak) [reply core/jazz_is-slack-channel-available/test/test.js]: A few other place in the code, you have similar code, so please update it in other places.

File: core/jazz_is-slack-channel-available/test/test.js [hunk 6]
```diff
@@ -1,11 +1,282 @@
...
+      index.requestToChannels(config, event.query.slack_channel)
+        .then(res => {
+          expect(res).to.have.property('is_available');
+          expect(res.is_available).to.be.true;
+          sinon.assert.calledTwice(getResponse);
+          getResponse.restore();
+        });
+    });
+
+    it("should indicate that provided slack_channel does not available in the user's list", () => {
```
Comment (suryajak) [inline core/jazz_is-slack-channel-available/test/test.js]: nitpick: "is not available"

File: core/jazz_is-slack-channel-available/test/test.js [hunk 7]
```diff
@@ -1,11 +1,282 @@
...
+        expect(res).to.have.deep.property('data.is_available');
+        expect(res.data.is_available).to.be.true;
+        sinon.assert.calledOnce(genericInputValidation);
+        sinon.assert.calledOnce(requestToChannels);
+        genericInputValidation.restore();
+        requestToChannels.restore();
+      });
+    });
+
+    it("should indicate that provided slack does not available", () => {
```
Comment (suryajak) [inline core/jazz_is-slack-channel-available/test/test.js]: nitpick: "is not available"

File: core/jazz_is-slack-channel-available/test/test.js (hunk 2 above)
Comment (supritat) [reply core/jazz_is-slack-channel-available/test/test.js]: Updated as per comment

File: core/jazz_is-slack-channel-available/test/test.js (hunk 3 above)
Comment (supritat) [reply core/jazz_is-slack-channel-available/test/test.js]: Updated as per comment

File: core/jazz_is-slack-channel-available/test/test.js (hunk 4 above)
Comment (supritat) [reply core/jazz_is-slack-channel-available/test/test.js]: Updated as per comment

File: core/jazz_is-slack-channel-available/test/test.js (hunk 5 above)
Comment (supritat) [reply core/jazz_is-slack-channel-available/test/test.js]: Updated as per comment

File: core/jazz_is-slack-channel-available/test/test.js (hunk 6 above)
Comment (supritat) [reply core/jazz_is-slack-channel-available/test/test.js]: Updated

File: core/jazz_is-slack-channel-available/test/test.js (hunk 7 above)
Comment (supritat) [reply core/jazz_is-slack-channel-available/test/test.js]: Updated

**Pull request merged**